src/main/java/org/cyclops/cyclopscore/config/configurable/ConfigurableBlockDoor.java

```diff
@@ -0,0 +1,86 @@
+package org.cyclops.cyclopscore.config.configurable;
+
+import net.minecraft.block.BlockDoor;
+import net.minecraft.block.SoundType;
+import net.minecraft.block.material.Material;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.client.renderer.color.IBlockColor;
+import net.minecraft.init.Items;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+import org.cyclops.cyclopscore.config.extendedconfig.BlockDoorConfig;
+
+import java.util.Random;
+
+import javax.annotation.Nullable;
+
+/**
+ * Door blockState that can hold ExtendedConfigs
+ * @author josephcsible
+ *
+ */
+public class ConfigurableBlockDoor extends BlockDoor implements IConfigurableBlock {
+    public Item item;
```

Comment:
Is this needed? Can't the item of this block be retrieved using `Item#getItemFromBlock`? Or is there another reason why this field is added here?

Reply:
Item#getItemFromBlock doesn't work right unless the item is an ItemBlock, which ItemDoor isn't (thanks, Mojang). Vanilla doors have to hack around it like this too.

Reply:
Oh ok, didn't know that, TIL.